src/client/mod.rs

```diff
@@ -285,9 +304,53 @@ impl KafkaClient {
     /// let mut client = kafka::client::KafkaClient::new(vec!("localhost:9092".to_owned()));
     /// client.load_metadata_all().unwrap();
     /// ```
+    pub fn new(hosts: Vec<String>) -> KafkaClient {
+        KafkaClient {
+            config: ClientConfig {
+                client_id: CLIENTID.to_owned(),
+                hosts: hosts,
+                compression: DEFAULT_COMPRESSION,
+                fetch_max_wait_time: DEFAULT_FETCH_MAX_WAIT_TIME,
+                fetch_min_bytes: DEFAULT_FETCH_MIN_BYTES,
+                fetch_max_bytes_per_partition: DEFAULT_FETCH_MAX_BYTES_PER_PARTITION,
+                fetch_crc_validation: DEFAULT_FETCH_CRC_VALIDATION,
+            },
+            conn_pool: ConnectionPool::new(DEFAULT_SO_TIMEOUT_SECS, None),
+            state: state::ClientState::new(),
+        }
+    }
+    
+    /// Creates a new secure instance of KafkaClient. Before being able to
+    /// successfully use the new client, you'll have to load metadata.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// extern crate openssl;
+    /// extern crate kafka;
+    /// use openssl::ssl::{Ssl, SslContext, SslStream, SslMethod, SSL_VERIFY_NONE};
+    /// use openssl::x509::X509FileType;
+    /// use kafka::client::{KafkaClient, SecurityConfig};
+    /// use std::path::Path;
+    ///
+    /// fn main() {
+    ///     let (key, cert) = ("foo.key".to_string(), "bar.crt".to_string());
+    ///
+    ///      // OpenSSL offers a variety of complex configurations. Here is an example:
+    ///      let mut context = SslContext::new(SslMethod::Sslv23).unwrap();
+    ///      context.set_cipher_list("DEFAULT").unwrap();
+    ///      context.set_certificate_file(&cert, X509FileType::PEM).unwrap();
+    ///      context.set_private_key_file(&key, X509FileType::PEM).unwrap();
+    ///      context.set_verify(SSL_VERIFY_NONE, None);
```

Comment:
btw: i think specifying `SSL_VERIFY_NONE` is a wrong/misleading example. a client surely wants to have the server certificate verified before sending any data.